docs/rules/no-unused-vars.md

```diff
@@ -195,6 +195,16 @@ Examples of **correct** code for the `{ "args": "none" }` option:
 })();
 ```
 
+### ignoreRestSiblings
+
+The `ignoreRestSiblings` option is a boolean (default: `false`). With modern JavaScript it is possible to "omit" properties from an object by using a rest parameter, but by default the sibling properties are marked as "unused". With this option enabled the omitted properties are considered "used", as their existence has a side-effect. In the example, `type` is considered "used".
```

Comment:
A few minor nitpicks:

* I feel like "modern JavaScript" is kind of a vague term, especially since rest/spread properties aren't part of the official spec yet. Maybe it would be better to link to the [rest/spread property spec](https://github.com/sebmarkbage/ecmascript-rest-spread) instead.
* I think this is technically referring to rest *properties*, not rest *parameters*.
* It might be clearer to say that the unused variables are ignored by the rule, rather than saying that the variables are considered "used" because their existence has a side-effect. For example, this variable's existence also has a side-effect, but it's not considered "used":

  ```js
  var foo = doSomethingWithSideEffects();
  ```